### ISSUE TITLE
tempdirectory (and maybe more in config.php defined folders) should be excluded from the check of extra files

### ISSUE THREAD

**Issue by eisenherzz** [nextcloud/server#14975]
**Is your feature request related to a problem? Please describe.**
When updating Nextcloud:
Check for expected files
The following extra files have been found:
tmp
This folder is defined via `'tempdirectory' => '/www/htdocs/cloud/tmp',`

**Describe the solution you'd like**
tempdirectory (and maybe more in config.php defined folders) should be excluded from the check of extra files

**Comment by yafp** [comment]
While upgrading to 15.0.7 i ran in that issue as well. The updater pre-check stopped because of the unexpected extra folder 'tmp'.
Removing that folder helped executing the upgrade. 

After having finished the upgrade the missing tmp folder was reported in the logging ;)

**Comment by eisenherzz** [comment]
As a workaround I moved the tmp folder outside the nextcloud
installation folder. This message should never come again ;)

**Comment by yafp** [comment]
Hey,
sounds like a possible workaround while i still think the problem should be properly solved ;)

**Comment by kesselb** [comment]
> Hey,
> sounds like a possible workaround while i still think the problem should be properly solved ;)

I don't agree with you. If you put the tmp directory inside document root it's reachable from the web (sure you can deny access to it) but the right way is to put it outside the document root.

**Comment by kesselb** [comment]
cc @MorrisJobke sounds like a nice feature but demand ...
